common/component/kafka/kafka.go

```diff
@@ -252,8 +283,12 @@ func (k *Kafka) Close() error {
 		}
 		k.subscribeLock.Unlock()
 
-		if k.cg != nil {
-			errs[1] = k.cg.Close()
+		if k.clients != nil {
+			errs[0] = k.clients.producer.Close()
+			errs[1] = k.clients.consumerGroup.Close()
```

Comment:
to keep semantics equal, set clients.producer and clients.consumerGroup to nil, apart from adding nil checks for them

if you notice there are nil checks for i.e clients.producer in several places

Reply:
@sicoyle